### PR TITLE
Fix Error code : 87

### PR DESCRIPTION
**Pull request by smallru8**
Win32API ReadFile WriteFile need OVERLAPPED parameter or remove FILE_FLAG_OVERLAPPED flag.
I have tested it in win10, and it can read and write tap device normally.

### CODE REVIEW

**Comment by Aversiste** [comment]
Hello,

Have you checked PR #20 regarding a better handling of overlapped io?

**Comment by rds13** [comment]
I don't expect visual code files to be usefull for this MR, aren't they ?

**Comment by Aversiste** [comment]
Hello @smallru8, can you remove the build32 and build64 folders? I don't think they are needed here and they take unnecessary space in the repository.

Files allowing libtuntap to be used in another language such as Java should be located in `bindings/java` just like we have folders for Python and C++.

I pushed two commits implementing just this in branch BounceGateVPN-master so you can cherry-pick them, otherwise you can allow me to modify your PR directly.

File: com_github_smallru8_driver_tuntap_TunTap.cpp [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+}
+
+/*
+ * Class:     com_github_smallru8_driver_tuntap_TunTap
+ * Method:    tuntap_read
+ * Signature: (I)[B
+ */
+JNIEXPORT jbyteArray JNICALL Java_com_github_smallru8_driver_tuntap_TunTap_tuntap_1read
+  (JNIEnv *env, jobject obj, jint len){
+	int ret = 512;
```
Comment (Aversiste) [inline com_github_smallru8_driver_tuntap_TunTap.cpp]: Reading your network packets by 512-bytes sized chunks each time is not going to end well.
Perhaps you should use the return value of `tuntap_get_readable` or `tuntap_get_mtu` instead.

File: com_github_smallru8_driver_tuntap_TunTap.cpp [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	if(len>0)
+		ret = len;
+	char *cData = (char*)malloc(sizeof(char)*ret);
```
Comment (Aversiste) [inline com_github_smallru8_driver_tuntap_TunTap.cpp]: Is this allocation freed by the JVM? Same question for Java_com_github_smallru8_driver_tuntap_TunTap_tuntap_1write.

**Comment by smallru8** [comment]
Hello @Aversiste , I'm sorry i forgot to delete build32 and build64 folders. I have deleted them.

I have also fixed and adjusted some problems. 
```
Fix read and write functions may cause memory leak.
Change reading network packets by 512-bytes to tuntap_get_mtu return value.
Move com_github_smallru8_driver_tuntap_TunTap and converter to bindings.
Add libtuntap4J java interface source for user to call.
```

**Comment by Aversiste** [comment]
Excellent, thank you.

If it is possible I would like to add an automatic test to our CI pipeline using your Test.java, but I fail to execute it:

```
$ /usr/local/jdk-11/bin/java Test.java
Test.java:13: error: cannot find symbol
        TunTap tt = new TunTap();
        ^
  symbol:   class TunTap
  location: class Test
Test.java:13: error: cannot find symbol
        TunTap tt = new TunTap();
                        ^
  symbol:   class TunTap
  location: class Test
2 errors
error: compilation failed
```

My Java skills are pretty much non existent, so what is going wrong?

**Comment by Aversiste** [comment]
Why are you closing this PR ? It seemed good.

**Pull request closed without merging**